File: src/main/java/com/vaadin/demo/component/stepper/StepperNavigation.java
package com.vaadin.demo.component.stepper;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.stepper.Step;
import com.vaadin.flow.component.stepper.Stepper;
import com.vaadin.flow.router.Route;

@Route("stepper-navigation")
public class StepperNavigation extends Div {

    public StepperNavigation() {
        // tag::snippet[]
        Stepper stepper = new Stepper();
        
        Step step1 = Step.withHref("Account setup", "#/account");
        step1.setState(Step.State.COMPLETED);
        
        Step step2 = Step.withHref("Profile information", "#/profile");
        step2.setState(Step.State.ACTIVE);
        
        Step step3 = Step.withHref("Preferences", "#/preferences");
        
        Step step4 = Step.withHref("Review", "#/review");
        
        stepper.add(step1, step2, step3, step4);
        // end::snippet[]

        add(stepper);
    }

    public static class Exporter extends DemoExporter<StepperNavigation> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/stepper/stepper-navigation.ts
import '@vaadin/stepper';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('stepper-navigation')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-stepper>
        <vaadin-step label="Account setup" href="#/account" state="completed"> </vaadin-step>
        <vaadin-step label="Profile information" href="#/profile" state="active"> </vaadin-step>
        <vaadin-step label="Preferences" href="#/preferences"> </vaadin-step>
        <vaadin-step label="Review" href="#/review"> </vaadin-step>
      </vaadin-stepper>
      <!-- end::snippet[] -->
    `;
  }
}
